File: data/src/main/java/org/cryptomator/data/cloud/local/file/RootLocalFolder.java
package org.cryptomator.data.cloud.local.file;

import android.os.Environment;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.LocalStorageCloud;

public class RootLocalFolder extends LocalFolder {

	private final LocalStorageCloud localStorageCloud;

	public RootLocalFolder(LocalStorageCloud localStorageCloud) {
		super(null, "", Environment.getExternalStorageDirectory().getPath());
		this.localStorageCloud = localStorageCloud;
	}

	@Override
	public Cloud getCloud() {
		return localStorageCloud;
	}

	@Override
	public RootLocalFolder withCloud(Cloud cloud) {
		return new RootLocalFolder((LocalStorageCloud) cloud);
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/local/file/RootLocalFolder.kt
package org.cryptomator.data.cloud.local.file

import android.os.Environment
import org.cryptomator.domain.Cloud
import org.cryptomator.domain.LocalStorageCloud

class RootLocalFolder(private val localStorageCloud: LocalStorageCloud) : LocalFolder(null, "", Environment.getExternalStorageDirectory().path) {

	override val cloud: Cloud
		get() = localStorageCloud

	override fun withCloud(cloud: Cloud?): RootLocalFolder {
		return RootLocalFolder(cloud as LocalStorageCloud)
	}
}
